### DISCUSSION TITLE
Error during migration

### DISCUSSION THREAD

**Discussion by ghost** [stashapp/stash#3043] [Q&A]
What does this mean and what do I have to do?

> An error occurred migrating the database to the latest schema version. The backup database file was automatically renamed to restore the database.
> error performing migration: Dirty database version 32. Fix and force version.
> 

[WP: restored deleted content]

**Comment by WithoutPants** [comment]
An error like this usually occurs if the migration process was interrupted when it was ran previously (crash/force quit/power outage as examples) and the database is now in an unknown state. If the migration process was interrupted, there should be a backup sqlite file from when the original migration was performed (since it only gets moved or removed when the migration process succeeds or fails).

Best bet is to restore the sqlite file from a previous backup if you have one. If not, you can clear the `dirty` flag with SQL:
```
UPDATE schema_migrations SET dirty = '0';
```

This way does not however guarantee that the migration will complete successfully, since the database schema may be partially migrated. If so, you will get a different error message, but we can walk through that if/when it occurs.
